editor/components/post-publish-button/label.js

```diff
@@ -28,7 +28,10 @@ export function PublishButtonLabel( {
 	isPublishing,
 	user,
 } ) {
-	const isContributor = user.data && ! user.data.capabilities.publish_posts;
+	const userCaps = user.data ?
+		{ ...user.data.capabilities, ...user.data.post_type_capabilities } :
+		{ publish_posts: false };
+	const isContributor = ! userCaps.publish_posts;
```

Comment:
Note to self: the test is probably failing because i don't test user.data anymore when setting isContributor.